hangman/game.py

```diff
@@ -0,0 +1,82 @@
+from word_generator import WordGenerator
+from constants import *
+from hangman import Hangman
+
+
+class Game:
+    def __init__(self):
+        self.word_generator = WordGenerator(SOWPODS)
+        self.word_generator.fetch_data()
+        self.hangman = None
+
+    def print_word(self) -> None:
+        print("\n        ", end='')
+        for i in self.hangman.dashed_word:
+            print(i, end=' ')
+        print('\n')
+
+    def get_input(self) -> str:
+        self.print_word()
+        print(self.hangman.chances_left, "Chances Left!!!")
+        print(self.hangman.hints_left, "Hints Left!!! (press 0 for hint)")
+        prompt: str = ">>>>> Guess your letter: "
+        inp_letter = input(prompt).upper()
+        return inp_letter
+
+    def check_input(self, inp: str) -> bool:
+        inp = inp.upper()
+        if(len(inp) != INPUT_LENGTH):
+            print("Enter only {:d} character".format(INPUT_LENGTH))
+            return False
+        # index of the alphabet entered [A: 0,B: 1 ,....]
+        alpha: int = ord(inp) - ord('A')
+
+        if(alpha < 0 or alpha >= NO_ALPHABETS):
+            print("Enter only alphabets(A-Z)")
+            return False
+
+        return True
+
+    def check_hint(self, inp: str) -> bool:
+        if(inp == HINT_OPTION):  # For hints
+            return True
+        return False
+
+    def get_endgame_input(self) -> str:
+        prompt: str = ">>>>> PRESS Q to End Game or Any Key to continue: "
+        inp: str = input(prompt)
+        return inp
+
+    def check_endgame(self, inp: str) -> bool:
+        if(inp == 'Q' or inp == 'q'):
+            return True
+        return False
+
+    def start_game(self) -> bool:
+        word: str = self.word_generator.get_word()
+        self.hangman = Hangman(word, HINTS_LEFT, CHANCES_LEFT)
```

Comment:
these two lines are initialising game, we can abstract it out for better readability
self.initialize_game()